provider/core/src/resource.rs

```diff
@@ -511,9 +526,39 @@ impl ResourceOptions {
         self.langid.region
     }
 
-    /// Returns the value of the specified Unicode extension keyword for this [`ResourceOptions`].
-    pub fn unicode_ext(&self, key: &unicode_ext::Key) -> Option<&unicode_ext::Value> {
-        self.keywords.get(key)
+    /// Gets the value of the specified Unicode extension keyword for this [`ResourceOptions`].
+    pub fn get_unicode_ext(&self, key: &unicode_ext::Key) -> Option<unicode_ext::Value> {
+        self.keywords.get(key).cloned()
+    }
+
+    /// Returns whether this [`ResourceOptions`] contains a Unicode extension keyword
+    /// with the specified key and value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_provider::prelude::*;
+    /// use icu_locid::{Locale, unicode_ext_key, unicode_ext_value};
+    ///
+    /// let locale: Locale = "it-IT-u-ca-coptic".parse()
+    ///     .expect("Valid BCP-47");
+    /// let options: ResourceOptions = locale.into();
+    ///
+    /// assert_eq!(
+    ///     options.get_unicode_ext(&unicode_ext_key!("hc")),
+    ///     None
+    /// );
+    /// assert_eq!(
+    ///     options.get_unicode_ext(&unicode_ext_key!("ca")),
+    ///     Some(unicode_ext_value!("coptic"))
+    /// );
+    /// assert!(options.matches_unicode_ext(
+    ///     &unicode_ext_key!("ca"),
+    ///     &unicode_ext_value!("coptic"),
+    /// ));
+    /// ```
+    pub fn matches_unicode_ext(&self, key: &unicode_ext::Key, value: &unicode_ext::Value) -> bool {
```

Comment:
nit: do we not usually pass tinystrs/subtags by value?

Reply:
I think the accepted convention is to pass Copy types by value; but there are exceptions both ways. At play here:

1. In this case, Key but not Value is Copy. I could pass one by value and the other by reference, or both by reference.
2. Many map interfaces pass by reference all the time, even for Copy types.

I don't have a strong opinion.